translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/project/ProjectProgress.java

```diff
@@ -0,0 +1,129 @@
+
+    Project project;
+    Context context;
+    ProjectDatabaseHelper db;
+
+    public ProjectProgress(Project project, Context context) {
+        this.project = project;
+        this.context = context;
+        this.db = new ProjectDatabaseHelper(context);
+    }
+
+    public int calculateProjectProgress() {
+        try {
+            ChunkPlugin chunks = project.getChunkPlugin(new ChunkPluginLoader(context));
+            int numChapters = chunks.numChapters();
+            int allChaptersProgress = 0;
+
+            List<Chapter> chapters = chunks.getChapters();
+            for (Chapter chapter: chapters) {
+                int chapterProgress = calculateChapterProgress(chapter);
+                allChaptersProgress += chapterProgress;
+            }
+            int projectProgress = (int) Math.ceil((float) allChaptersProgress / numChapters);
+
+            return projectProgress;
+        } catch (IOException e) {
+            Logger.i(this.toString(), e.getMessage());
+        }
+
+        return 0;
+    }
+
+    public void setProjectProgress(int progress) {
+        try {
+            int projectId = db.getProjectId(project);
+            db.setProjectProgress(projectId, progress);
+        } catch (IllegalArgumentException e) {
+            Logger.i(this.toString(), e.getMessage());
+        }
+    }
+
+    public void updateProjectProgress() {
+        int projectProgress = calculateProjectProgress();
+        setProjectProgress(projectProgress);
+    }
+
+    public int calculateChapterProgress(Chapter chapter) {
+        int unitCount = chapter.getChunks().size();
+        int chapterNumber = chapter.getNumber();
+        Map<Integer, Integer> unitsStarted = db.getNumStartedUnitsInProject(project);
+        if(unitsStarted.get(chapterNumber) != null) {
+            int chapterProgress = calculateProgress(unitsStarted.get(chapterNumber), unitCount);
+
+            return chapterProgress;
+        }
+
+        return 0;
+    }
+
+    public void setChapterProgress(Chapter chapter, int progress) {
+        try {
+            int chapterId = db.getChapterId(project, chapter.getNumber());
+            db.setChapterProgress(chapterId, progress);
+        } catch (IllegalArgumentException e) {
+            Logger.i(this.toString(), e.getMessage());
+        }
+    }
+
+    public void updateChapterProgress(Chapter chapter) {
+        int chapterProgress = calculateChapterProgress(chapter);
+        setChapterProgress(chapter, chapterProgress);
+    }
+
+    public void updateChaptersProgress() {
+        try {
+            ChunkPlugin chunks = project.getChunkPlugin(new ChunkPluginLoader(context));
+            List<Chapter> projectChapters = chunks.getChapters();
+            for (Chapter chapter: projectChapters) {
+                updateChapterProgress(chapter);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Chapter getChapter(int chapterNumber) {
```

Comment:
this really shouldn't be a public API for a progress use case object

Reply:
@sarabiaj Updated!